frontends/main/src/app-pages/DashboardPage/CoursewareDisplay/transform.ts

```diff
@@ -17,16 +17,26 @@ import { groupBy } from "lodash"
 const sources = {
   mitxonline: "mitxonline",
 }
-const getId = (
-  source: string,
-  resourceType: DashboardResourceType,
-  id: number,
-) => `${source}-${resourceType}-${id}`
+type KeyOpts = {
+  source: string
+  resourceType: DashboardResourceType
+  id: number
+  runId?: number
+}
+const getKey = ({ source, resourceType, id, runId }: KeyOpts) => {
+  const base = `${source}-${resourceType}-${id}`
+  return runId ? `${base}-${runId}` : base
+}
```

Comment:
Adding run-id here fixes a react duplicate key error when you're enrolled in two runs of the same course.